### PR TITLE
[REVIEW] cudf documentation spell check fix

### LINKED ISSUE #5154: [DOC] spell check on documentation text
## Report incorrect documentation

**Location of incorrect documentation**

Spelling/Typo errors in documentation.
For example,
https://docs.rapids.ai/api/libcudf/nightly/namespacecudf_1_1experimental.html#a7e8216839cc624ea3a69aa60943ccbde
https://github.com/rapidsai/cudf/blob/cdefa41a6656579b36875abbc6c7f452d67139cf/cpp/include/cudf/unary.hpp#L70

**Describe the problems or issues found in the documentation**
For example, `non-nulalble` should be `non-nullable`. 
Similar misspellings should be fixed

**Suggested fix for documentation**
run spell check and fix the misspellings.

### PR DESCRIPTION
**Pull request by karthikeyann**
closes #5154 
- [x] libcudf documentation spell check (excluding legacy)
- [x] python doc string spell check

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by davidwendt** [comment]
This is amazing. How did you run spell check on the source code?

**Comment by harrism** [comment]
I use the "code spell check" vscode extension.

**Comment by karthikeyann** [comment]
I read the docs of [Code Spell Checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) and used its backend **[Cspell](https://www.npmjs.com/package/cspell)** in command line and analysed on all C++ and CUDA files (filtered comments only). Fixes are manual.

File: python/cudf/cudf/core/column/numerical.py [hunk 1]
```diff
@@ -222,7 +222,7 @@ def round(self, decimals=0):
         return column.build_column(data=data, dtype=self.dtype, mask=self.mask)
 
     def applymap(self, udf, out_dtype=None):
-        """Apply a elemenwise function to transform the values in the Column.
+        """Apply a elementwise function to transform the values in the Column.
```
Comment (cwharris) [inline python/cudf/cudf/core/column/numerical.py:225]:
```suggestion
        """Apply an element-wise function to transform the values in the Column.
```

File: python/cudf/cudf/core/column/column.py [hunk 2]
```diff
@@ -1578,7 +1578,7 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
 
 
 def column_applymap(udf, column, out_dtype):
-    """Apply a elemenwise function to transform the values in the Column.
+    """Apply a elementwise function to transform the values in the Column.
```
Comment (cwharris) [inline python/cudf/cudf/core/column/column.py:1581]:
```suggestion
    """Apply an element-wise function to transform the values in the Column.
```

File: python/cudf/cudf/comm/serialize.py [hunk 3]
```diff
@@ -30,7 +30,7 @@ def cuda_serialize_cudf_object(x):
             header["lengths"] = [f.nbytes for f in frames]
             return header, frames
 
-    # all (de-)serializtion are attached to cudf Objects:
+    # all (de-)serialization are attached to cudf Objects:
```
Comment (cwharris) [inline python/cudf/cudf/comm/serialize.py:33]:
```suggestion
    # all (de-)serializations are attached to cudf Objects:
```

File: python/cudf/cudf/comm/serialize.py [hunk 4]
```diff
@@ -3,7 +3,7 @@
 import cudf
 import cudf.core.groupby.groupby
 
-# all (de-)serializtion are attached to cudf Objects:
+# all (de-)serialization are attached to cudf Objects:
```
Comment (cwharris) [inline python/cudf/cudf/comm/serialize.py:6]:
```suggestion
# all (de-)serializations are attached to cudf Objects:
```

File: python/cudf/cudf/_lib/string_casting.pyx [hunk 5]
```diff
@@ -484,7 +484,7 @@ def htoi(Column input_col, **kwargs):
 
     Returns
     -------
-    A Column with strings representd as integer
+    A Column with strings represented as integer
```
Comment (cwharris) [inline python/cudf/cudf/_lib/string_casting.pyx:487]:
```suggestion
    A Column of integers parsed from hexadecimal string values.
```

File: cpp/src/jit/parser.h [hunk 6]
```diff
@@ -118,7 +118,7 @@ class ptx_parser {
    * @brief Convert the input PTX instruction into a inline PTX
```
Comment (cwharris) [inline cpp/src/jit/parser.h:118]:
```suggestion
   * @brief Convert the input PTX instruction into an inline PTX
```

File: cpp/src/jit/parser.h [hunk 7]
```diff
@@ -118,7 +118,7 @@ class ptx_parser {
...
    * statement without changing (exceptions exist).
    *
-   * Non-alphanemric that are not underscore in register identifiers
+   * Non-alphanumeric that are not underscore in register identifiers
    * are all replaced with underscore. An example:
```
Comment (cwharris) [inline cpp/src/jit/parser.h:122]:
```suggestion
   * Non-alphanumeric characters in register identifiers, except underscores, are replaced with
   underscore. Example:
```

File: cpp/src/io/csv/csv_common.h [hunk 8]
```diff
@@ -36,7 +36,7 @@ enum : uint8_t {
 using flags = uint8_t;
 
 /**
- * @brief Per-column histogram struct holding detected ocurrences of each dtype
+ * @brief Per-column histogram struct holding detected occurrences of each dtype
```
Comment (cwharris) [inline cpp/src/io/csv/csv_common.h:39]:
```suggestion
 * @brief Per-column histogram struct containing detected occurrences of each dtype
```

File: cpp/src/hash/concurrent_unordered_map.cuh [hunk 9]
```diff
@@ -310,7 +310,7 @@ class concurrent_unordered_map {
    * that the insert did not succeed.
    *
    * If the new key was not present, the iterator points to the location
-   *where the insert occured and the boolean is `true` indicating that the
+   *where the insert occurred and the boolean is `true` indicating that the
```
Comment (cwharris) [inline cpp/src/hash/concurrent_unordered_map.cuh:313]:
```suggestion
   * where the insert occurred and the boolean is `true` indicating that the
```

File: cpp/src/bitmask/null_mask.cu [hunk 10]
```diff
@@ -523,7 +523,7 @@ std::vector<size_type> segmented_count_set_bits(bitmask_type const *bitmask,
 
   CHECK_CUDA(stream);
 
-  // third adjust counts in segement boundaries (if segments are not
+  // third adjust counts in segment boundaries (if segments are not
```
Comment (cwharris) [inline cpp/src/bitmask/null_mask.cu:526]:
```suggestion
  // third, adjust counts in segment boundaries (if segments are not
```

File: cpp/include/cudf/detail/reduction_operators.cuh [hunk 11]
```diff
@@ -111,7 +111,7 @@ struct simple_op {
 };
 
 // `sum`, `product`, `sum_of_squares`, `min`, `max` are used at simple_reduction
-// inferface is defined by CRTP calss simple_op
+// inferface is defined by CRTP class simple_op
```
Comment (harrism) [inline cpp/include/cudf/detail/reduction_operators.cuh:114]: This one didn't get fixed!

Comment (karthikeyann) [inline cpp/include/cudf/detail/reduction_operators.cuh:114]:
```suggestion
// interface is defined by CRTP class simple_op
```

**Comment by karthikeyann** [comment]
> Pretty fun to read all typos in the project at once :D
> 
> I noticed some types are related to C++ names. Does the tool automatically detect these?

C++ keywords are automatically detected. For some reason, C++14 keywords were not present. So added my own dictionary from cppreference.com keywords.

File: python/cudf/cudf/core/dataframe.py [hunk 12]
```diff
@@ -1391,9 +1391,8 @@ def index(self, value):
         if isinstance(value, cudf.core.multiindex.MultiIndex):
             if len(self._data) > 0 and len(value) != len(self):
                 msg = (
-                    f"Length mismatch: Expected axis has "
-                    "%d elements, new values "
-                    "have %d elements" % (len(self), len(value))
+                    f"Length mismatch: Expected axis has {len(self)} elements,"
```
Comment (cwharris) [inline python/cudf/cudf/core/dataframe.py]: Do these `f`s need to be removed? (other files as well).

Comment (kkraus14) [reply python/cudf/cudf/core/dataframe.py]: No, f-strings are a python thing that allow you to use the `{ }` notation to use python expressions inside of strings.

Comment (harrism) [reply python/cudf/cudf/core/dataframe.py]: Other changes in this same commit removed the f from strings in categorical.py and column.py. @karthikeyann please verify.

Comment (harrism) [reply python/cudf/cudf/core/dataframe.py]: Ah, I guess f was not required in those instances due to no `{ }` notation.

Comment (karthikeyann) [reply python/cudf/cudf/core/dataframe.py]: Yes. f-strings without `{ }` are flagged as warning in flake8 `F541 f-string is missing placeholders`. So f was removed in those cases. 
This line has `{ }` placeholder. so, f is needed here.

**Pull request merged**